pkg/services/notification/resource_control.go

```diff
@@ -134,3 +138,195 @@ func GetEmailServiceConfig(ctx context.Context) *pb.EmailServiceConfig {
 
 	return emailCfg
 }
+
+func DescribeNotifications(req *pb.DescribeNotificationsRequest) ([]*models.Notification, uint64, error) {
+
+	var nfs []*models.Notification
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
+
+	var count uint64
+	var err error
+	db, displayColumns, orderByStr := buildDB4DescNfs(req)
+
+	nfs = make([]*models.Notification, 0)
+	err = db.Select(displayColumns).
+		Offset(offset).
+		Limit(limit).
+		Order(orderByStr).
+		Find(&nfs).
+		Count(&count).
+		Error
+	if err != nil {
+		logger.Errorf(nil, "Failed to describe notifications [%v], error: %+v.", req, err)
+		return nil, 0, err
+	}
+
+	if req.DisplayColumns != nil && req.DisplayColumns[0] == "" {
+		return make([]*models.Notification, 0), count, err
+	}
+
+	return nfs, count, err
+
+}
+func buildDB4DescNfs(req *pb.DescribeNotificationsRequest) (*gorm.DB, []string, string) {
+	//Step1：get Filter
+	db := globalcfg.GetInstance().GetDB()
```

Comment:
db = db, will cause origin db be changed. 

you can init:
db := globalcfg.GetInstance().GetDB().Table(constants.TableNotification)

In .Table will clone new db.